packages/worker/src/workflow/vm.ts

```diff
@@ -163,7 +169,24 @@ export class VMWorkflow implements Workflow {
       // Wait for microtasks to be processed
       await new Promise((resolve) => process.nextTick(resolve));
     }
-    return this.workflowModule.concludeActivation();
+    const ret = this.workflowModule.concludeActivation();
+    // Give unhandledRejection handler a chance to process.
+    // Apparently nextTick does not get it triggered so we use setTimeout here.
+    await new Promise((resolve) => setTimeout(resolve, 0));
```

Comment:
Seems odd. Maybe we could set a flag in the handler, and run next tick until we see the flag set (then unset it again) ?

Reply:
I think it might not be enough